src/handlers/thirdparty.ts

```diff
@@ -220,3 +250,56 @@ export function handleItemReviewed(event: ItemReviewed): void {
 
   item.save()
 }
+
+let currentReceiptId = 0
```

Comment:
The receipt needs an id so I'm using this as an increasing counter. Kinda ugly but simple. Any better idea is welcome.

Reply:
we could use the `event.transaction.hash.toHex()` ? or maybe the block hash/number/block     
```ts
    event.block.timestamp
    event.block.hash.toHexString()
    event.transaction.hash.toHexString()
```

Reply:
More than one of these events might be emitted on the same transaction. 

```
    function _consumeSlots(string calldata _thirdPartyId, ConsumeSlotsParam[] calldata _consumeSlotsParams) internal {
        address sender = _msgSender();

        ThirdParty storage thirdParty = thirdParties[_thirdPartyId];

        for (uint256 i = 0; i < _consumeSlotsParams.length; i++) {
            ConsumeSlotsParam memory consumeSlotParam = _consumeSlotsParams[i];

            require(consumeSlotParam.qty > 0, "TPR#_consumeSlots: INVALID_QTY");

            uint256 newConsumedSlots = thirdParty.consumedSlots.add(consumeSlotParam.qty);

            require(thirdParty.maxItems >= newConsumedSlots, 'TPR#_consumeSlots: NO_ITEM_SLOTS_AVAILABLE');

            bytes32 messageHash = toTypedMessageHash(
                keccak256(abi.encode(CONSUME_SLOTS_TYPEHASH, keccak256(bytes(_thirdPartyId)), consumeSlotParam.qty, consumeSlotParam.salt))
            );

            require(thirdParty.receipts[messageHash] == 0, 'TPR#_consumeSlots: MESSAGE_ALREADY_PROCESSED');

            address signer = ecrecover(messageHash, consumeSlotParam.sigV, consumeSlotParam.sigR, consumeSlotParam.sigS);

            require(thirdParty.managers[signer], 'TPR#_consumeSlots: INVALID_SIGNER');

            thirdParty.receipts[messageHash] = consumeSlotParam.qty;
            thirdParty.consumedSlots = newConsumedSlots;

            emit ItemSlotsConsumed(_thirdPartyId, consumeSlotParam.qty, signer, sender);
        }
    }
```

So using anything block or transaction specific might not be enough.

Reply:
I think the solution could come from the Counter entity.

Instead of:
```
const receipt = new Receipt((++currentReceiptId).toString())
const metric = buildCountFromReceipt()
metric.save()
```

I could
```
const metric = buildCountFromReceipt()
metric.save()
  
const receipt = new Receipt(metric.receiptTotal.toString())
```